### PR TITLE
Add Preliminary Support for Linter

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

File: main/src/library/BigInt.flix [hunk 1]
```diff
@@ -93,7 +93,62 @@ namespace BigInt {
...
+        forall (x: BigInt).
+            (1ii * x) === x
+
+    @lint
+    def rightMultiplicationByOne(): Bool =
+        forall (x: BigInt).
+            (x * 1ii) === x
+
+    @lint
+    def divisionByOne(): Bool =
```
Comment (mlutze) [inline main/src/library/BigInt.flix]: Should we include division by zero as well?

Comment (magnus-madsen) [reply main/src/library/BigInt.flix]: Yeah, I guess we could :)

**Comment by acieroid** [comment]
I can't push to the branch, but here is a pattern that we may want to express:     // Here's a pattern that we may want to express (but we can't): `Vector.exists(f, v)` followed by `Vector.find(v, f)` can always be replaced by only `Vector.find(v, f)`.
```
@lint
def existsFindLeft(): Bool =
  forall [a] (f: a -> Bool, v: Vec[a]).
    (Vector.exists(f, v) `followedBy` Vector.find(v, f))
    === // not really ===, as the context has to change
    Vector.find(v, f)
```

And here are some patterns for an hypothetical image library:
```
enum Image {
  // Not defined
}

enum Color {
  // Not defined
}

namespace Image {
  // Returns a new blank image
  pub def blank(width: Int, height: Int): Image = ???

  // Returns a new colored image
  pub def colored(width: Int, height: Int, color: Color): Image = ???

  // Copies an image
  pub def copy(img: Image): Image = ???

  // Fills an image
  pub def fill(image: mut Image, color: Color): Unit = ???

  // Resizes an image
  pub def resize(image: mut Image, width: Int, height: Int): Unit = ???

  // Overlays an image on top of another
  pub def overlay(image: mut Image, overlay: Image, x: Int, y: Int): Unit = ???

  // Clears a section of an image with the given color
  pub def clear(image: mut Image, color: Color, x: Int, y: Int, w: Int, h: Int): Unit = ???

  // Filling a blank image to obtain a colored one can be replaced by using `colored`
  @lint
  pub def blankFill(): Bool =
    forall (img: Image, color: Color, w: Int, h: Int).
      (img == blank(w, h))
      `implies`
      (fill(copy(img), color) == colored(w, h, color))

  // Two resizes in a row can be replaced by a single resize operation
  @lint
  pub def resizeTwice(): Bool =
    forall (img: Image, w1: Int, h1: Int, w2: Int, h2: Int).
      (resize(img, w1, h1) `followedBy` resize(img, w2, h2))
      ==
      resize(img, w2, h2)

  @lint
  pub def overlayColorIsClear(): Bool =
    forall (img: Image, color: Color, x: Int, y: Int, w: Int, h: Int).
      overlay(img, colored(w, h, color), x, y) == clear(img, color, x, y, w, h)
}
```

**Comment by magnus-madsen** [comment]
@acieroid I am going to close this PR soon (merge), so to makes sure your comment does not get lost, can we put it somewhere else? Perhaps in the paper or something?

**Pull request merged**